### DISCUSSION TITLE
C++ Documentation for Mitsuba and Dr.JIT?

### DISCUSSION THREAD

**Discussion by kimren227** [mitsuba-renderer/mitsuba3#306] [Q&A] [answered]
Hi,

I'm developing C++ plugins for Mitsuba3 and I find it quite difficult (for scalar variant is manageable, but not so easy for jit related functions) as no C++ reference for Dr.Jit is present. Currently, I'm partially relying on enoki and Mitsuba2's C++ reference, it helps for the data structure parts, however, there still are differences between the API and underlying mechanism.  

Is there a C++ reference for Dr.Jit and Mitsuba3?

Best Regards,

Daxuan

**Comment by Speierers** [comment, marked as answer]
Unfortunately, the Dr.Jit documentation is still WIP. However, for both frameworks the the C++ and Python APIs are very similar so you should be able to look at the API reference pages for references.

[Mitsuba 3 API reference](https://mitsuba.readthedocs.io/en/stable/src/api_reference.html) 
[Dr.Jit API reference](https://drjit.readthedocs.io/en/latest/reference.html)

**Reply by kimren227** [reply]
Thanks for the info, is there a release schedule for the C++ documentation?

**Reply by Speierers** [reply]
On the Mitsuba side, we are not planning on providing any C++ documentation as the Python API is very similar. For Dr.Jit can't really give you an ETA unfortunately.

**Reply by kimren227** [reply]
No worries and Thank you! Looking forward to it.
